scipy/optimize/_lsq/common.py

```diff
@@ -0,0 +1,666 @@
+"""Functions used by least-squares algorithms."""
+from math import copysign
+
+import numpy as np
+from numpy.linalg import norm
+
+from scipy.linalg import cho_factor, cho_solve, LinAlgError
+from scipy.sparse import issparse
+from scipy.sparse.linalg import LinearOperator, aslinearoperator
+
+
+EPS = np.finfo(float).eps
+
+
+# Functions related to a trust-region problem.
+
+
+def intersect_trust_region(x, s, Delta):
+    """Find the intersection of a line with a spherical bound of a trust
```

Comment:
That's why writing short summaries is hard. It's good brain exercise though ;) Maybe
```
"""Find the intersection of a line with a trust region bound. 

Blah
"""